jerry-core/ecma/operations/ecma-typedarray-object.c

```diff
@@ -1184,21 +1171,236 @@ ecma_op_typedarray_from (ecma_value_t items_val, /**< the source array-like obje
+  }
+
+  /* 5 */
+  ecma_value_t using_iterator = ecma_op_get_method_by_symbol_id (source_val, LIT_GLOBAL_SYMBOL_ITERATOR);
+
+  if (ECMA_IS_VALUE_ERROR (using_iterator))
+  {
+    return using_iterator;
+  }
+
+  /* 6 */
+  if (!ecma_is_value_undefined (using_iterator))
+  {
+    /* 6.a */
+    ecma_value_t next_method;
+    ecma_value_t iterator = ecma_op_get_iterator (source_val, using_iterator, &next_method);
+    ecma_free_value (using_iterator);
+
+    /* 6.b */
+    if (ECMA_IS_VALUE_ERROR (iterator))
+    {
+      return iterator;
+    }
+
+    /* 6.c */
+    ecma_collection_t *values_p = ecma_new_collection ();
+    ecma_value_t ret_value = ECMA_VALUE_EMPTY;
+
+    /* 6.e */
+    while (true)
+    {
+      ecma_value_t next = ecma_op_iterator_step (iterator, next_method);
+
+      if (ECMA_IS_VALUE_ERROR (next))
+      {
+        ret_value = next;
+        break;
+      }
+
+      if (next == ECMA_VALUE_FALSE)
+      {
+        break;
+      }
+
+      ecma_value_t next_value = ecma_op_iterator_value (next);
+      ecma_free_value (next);
+
+      if (ECMA_IS_VALUE_ERROR (next_value))
+      {
+        ret_value = next_value;
+        break;
+      }
+
+      ecma_collection_push_back (values_p, next_value);
+    }
+
+    ecma_free_value (iterator);
+    ecma_free_value (next_method);
+
+    if (ECMA_IS_VALUE_ERROR (ret_value))
+    {
+      ecma_collection_free (values_p);
+      return ret_value;
+    }
+
+    /* 6.c */
+    ecma_object_t *constructor_obj_p = ecma_get_object_from_value (this_val);
+    ecma_value_t len_val = ecma_make_uint32_value (values_p->item_count);
+    ecma_value_t new_typedarray = ecma_typedarray_create (constructor_obj_p, &len_val, 1);
+
+    if (ECMA_IS_VALUE_ERROR (new_typedarray))
     {
-      ecma_value_t set_value = ecma_op_typedarray_from_helper (this_val,
-                                                               current_value,
+      ecma_collection_free (values_p);
+      return new_typedarray;
+    }
+
+    ecma_object_t *new_typedarray_p = ecma_get_object_from_value (new_typedarray);
+    ecma_typedarray_info_t info = ecma_typedarray_get_info (new_typedarray_p);
+    ecma_typedarray_setter_fn_t setter_cb = ecma_get_typedarray_setter_fn (info.id);
+
+    ret_value = ecma_make_object_value (new_typedarray_p);
+
+    /* 6.e */
+    for (uint32_t index = 0; index < values_p->item_count; index++)
+    {
+      ecma_value_t set_value = ecma_op_typedarray_from_helper (this_arg,
+                                                               values_p->buffer_p[index],
                                                                index,
                                                                func_object_p,
                                                                &info,
                                                                setter_cb);
 
       if (ECMA_IS_VALUE_ERROR (set_value))
       {
+        for (uint32_t j = index + 1; j < values_p->item_count; j++)
```

Comment:
are you sure that the current element should not be released?

Reply:
yes since `ecma_op_typedarray_from_helper` free the current element